pkg/util/kubernetes/autoscalers/datadogexternal_util.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoscalers
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	le "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/metrics"
+)
+
+type minRemainingRequests struct {
+	sync.Mutex
+	val            int
+	timestamp      time.Time
+	expiryDuration time.Duration
+}
+
+func newMinRemainingRequests(expiryDuration time.Duration) minRemainingRequests {
+	return minRemainingRequests{
+		val:            -1,
+		timestamp:      time.Now(),
+		expiryDuration: expiryDuration,
+	}
+}
+
+func (mrr *minRemainingRequests) update(newVal string) {
+	mrr.Lock()
+	defer mrr.Unlock()
+
+	newValFloat, err := strconv.Atoi(newVal)
```

Comment:
You get an `int`, not a `float`, it's usually not necessary to suffix with type like `newVal` is enough